libmscore/text.cpp

```diff
@@ -1044,13 +1044,7 @@ QRectF Text::cursorRect() const
 
 QColor Text::textColor() const
       {
-      if (score() && !score()->printing()) {
-            if (selected())
-                  return (track() > -1) ? MScore::selectColor[voice()] : MScore::selectColor[0];
-            if (!visible())
-                  return Qt::gray;
-            }
-      return textStyle().foregroundColor();
+      return curColor();
```

Comment:
That doesn't work. It will change the behavior since you don't call textStyle().foregroundColor(); anymore.

I would suggest that you focus on the master branch instead of 2.2 if it's your first contribution to OS. It's great that you managed to compile MuseScore. If you need more help, you can talk with other developers on IRC #musescore on freenode.net https://kiwiirc.com/client/irc.freenode.net/?nick=musescore|?#musescore

Reply:
Thank you for the tips and feed backs! I will come back to work on this after I have gained a better understanding of the project!